File: src/ShippingService.java
import java.util.List;

public class ShippingService {
    private static final double SHIPPING_RATE_PER_KG = 30.0; // 30 per kg
    
    public static void ship(List<Shippable> shippableItems) {
        if (shippableItems.isEmpty()) {
            return;
        }
        
        System.out.println("** Shipment notice **");
        double totalWeight = 0.0;
        
        for (Shippable item : shippableItems) {
            System.out.println("1x " + item.getName() + " " + (int)(item.getWeight() * 1000) + "g");
            totalWeight += item.getWeight();
        }
        
        System.out.println("Total package weight " + totalWeight + "kg");
    }
    
    public static double calculateShippingFee(List<Shippable> shippableItems) {
        if (shippableItems.isEmpty()) {
            return 0.0;
        }
        
        double totalWeight = shippableItems.stream()
                .mapToDouble(Shippable::getWeight)
                .sum();
        
        return totalWeight * SHIPPING_RATE_PER_KG;
    }
}

File: python/shipping_service.py
class ShippingService:
    SHIPPING_RATE_PER_KG = 30.0

    @staticmethod
    def ship(shippable_items):
        if not shippable_items:
            return

        print("** Shipment notice **")
        total_weight = 0.0

        for item in shippable_items:
            print(f"1x {item.get_name()} {int(item.get_weight() * 1000)}g")
            total_weight += item.get_weight()

        print(f"Total package weight {total_weight}kg")

    @staticmethod
    def calculate_shipping_fee(shippable_items):
        if not shippable_items:
            return 0.0

        total_weight = sum(item.get_weight() for item in shippable_items)
        return total_weight * ShippingService.SHIPPING_RATE_PER_KG
